compiler/src/iree/compiler/Codegen/ExternalInterfaces/CodegenExternalModels.cpp

```diff
@@ -0,0 +1,65 @@
+// Copyright 2025 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "iree/compiler/Codegen/ExternalInterfaces/CodegenExternalModels.h"
+
+#include "iree/compiler/Codegen/Dialect/Codegen/IR/IREECodegenAttrs.h"
+#include "iree/compiler/Codegen/Dialect/Codegen/IR/IREECodegenDialect.h"
+#include "iree/compiler/Dialect/Encoding/IR/EncodingTypes.h"
+#include "iree/compiler/Dialect/TensorExt/IR/TensorExtTypes.h"
+
+namespace mlir::iree_compiler::IREE::Codegen {
+
+using IREE::TensorExt::DispatchTensorType;
+
+struct EncodingNopHostEncodingLayoutResolverAttrInterface final
+    : IREE::Encoding::EncodingLayoutResolverAttrInterface::ExternalModel<
+          EncodingNopHostEncodingLayoutResolverAttrInterface,
+          EncodingNopLayoutAttr> {
+  Attribute cloneWithSimplifiedConfig(Attribute attr,
+                                      DictionaryAttr config) const {
+    return EncodingNopLayoutAttr::get(attr.getContext());
```

Comment:
Do we need to create a new attribute? Can we return `attr` directly?

Reply:
Yes, done